### ISSUE TITLE
Proposal: boolean operators in plain English

### ISSUE THREAD

**Issue by laurentpayot** [gren-lang/compiler#173]
Hi, Gren looks so amazing that I’m considering migrating my project from Elm to Gren. Gren is such a breath of fresh air, so a big thank to the all the contributors :+1: 

Here is an [Elm proposal](https://github.com/elm/compiler/issues/2031) that was closed, and that I’m proposing to Gren too. I will not be offended if not accepted too :wink: 

<hr>

When I started programming in Elm I was a bit confused by the boolean operators syntax that is mixing C language legacy symbols like `&&` and `||` and English words like `not` and `xor`, as well as the Haskell-like `/=`. Not to mention that equality `==` is sometimes hard to distinguish from assignment `=` when using a ligaturized font.

Since Gren is still a work in progress I was wondering if CoffeeScript-inspired plain English operators `and`, `or`, `is` and `isnt` could be introduced in the compiler so we could turn things like
```elm
myBool = foo == 1 && bar /= 2 || not baz
```
into
```coffee
myBool = foo is 1 and bar isnt 2 or not baz
```

What are your thoughts?

**Comment by lue-bird** [comment]
Like the idea! Seems a bit much to make all of those reserved words...
- They are a lot less "important" in comparison to all the other keywords that rather describe structure, like `import`, `if`, `module`
- It's not really obvious what is an argument and what is a function
An alternative suggestions
- Make them just normal functions, like `foo |> is 1 |> and (bar |> isnt 2) |> or (not baz)` which would pair well with custom helpers etc. and makes order etc. more obvious https://github.com/gren-lang/compiler/issues/79
- reintroduce ability to make normal core functions infix-able, like `foo ´is 1 ´and bar ´isnt 2 ´or ´not`

**Comment by robinheghan** [comment]
Hello @laurentpayot 👋 Thank you for taking an interest in the Gren project!

There's nothing terribly wrong with this proposal, I've used both Ruby and CoffeeScript in the past and like not having to contort my fingers to write those symbols, but I'm still going to reject this for the following two reasons:

1. It would make it harder to differantiate between operators and functions. Currently, all functions are made up of readable text while operators uses symbols. Being able to tell, at a glance, what works with infix arguments and what works with suffix arguments help readability. Also, lue has a good point about the increase in reservered words being unfortunate.

2. (and this is the main one) This is going to require a big rewrite of all existing Gren code, and I don't think this change is worth the hours that'll take. I'm certain someone will wastly prefer this way of writing boolean logic, but many won't see the point, and will be irritated by the re-write.

**Comment by laurentpayot** [comment]
@robinheghan no worries :wink:
